src/main/java/io/cdap/plugin/format/MultiTableConf.java

```diff
@@ -175,11 +187,22 @@ public Connection getConnection() throws SQLException {
   }
 
   public String appendWhereClause(String selectClause, String table) {
-    StringBuilder query = new StringBuilder(selectClause).append(table);
+    String query = selectClause + table;
+    if (whereClause != null && !whereClause.isEmpty()) {
+      query += " " + whereClause;
+    }
+
+    return query;
+  }
+
+  public String appendWhereClauseWithSplit(String selectClause, String table, String split) {
```

Comment:
this doesn't belong in the config. The query is logic internal to the input format.

Reply:
Moved  appendWhereClauseWithSplit() logic to DBTableRecordReader and appendWhereClause() logic to MultiTableDBInputFormat class.